### PR TITLE
k_retrieval function to find the k obj of a list with more similarity…

### PR DESCRIPTION
**Pull request by mrfarge**
… with the query

### CODE REVIEW

File: src/k_retrieval.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+def k_retrieval(distances_list, similarity_methods, k):
```
Comment (hermesvf) [inline src/k_retrieval.py]: It is not clear to me what do each argument represent (other than `k`).

File: src/k_retrieval.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    
+    """
+    
+    Returns the K more similar images of the database from a list of distance obj
+    
+    """
+    
+    method=distances_list[0].method
```
Comment (hermesvf) [inline src/k_retrieval.py]: why is it a list and not a single distance?

File: src/k_retrieval.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    if method == "euclidean" or method == "L1_dist" or method == "x2_dist" or method == "hellinger":
```
Comment (hermesvf) [inline src/k_retrieval.py]: The `if` and the `else` have almost the same code. I would change to something like 
```suggestion
    maximization_required = method not in [ "euclidean", "L1_dist", "x2_dist", "hellinger"]
    distances_list.sort(key=lambda x: x.distance, reverse=maximization_required)
    k_distances=distances_list[0 : k+1]
    

```

Comment (hermesvf) [reply src/k_retrieval.py]: Even better, I would store in the "distance" class if the metric needs maximization or minimization.

File: src/k_retrieval.py (hunk 2 above)
Comment (mrfarge) [reply src/k_retrieval.py]: I've assumed a list of distances as input, so I think it's easier to rank the distance. Do you think would be better do it different?

File: src/k_retrieval.py (hunk 3 above)
Comment (mrfarge) [reply src/k_retrieval.py]: Yep, you're right, I'm changing it.

**Pull request closed without merging**